app/forms/household_relationship_form.py

```diff
@@ -17,16 +18,17 @@ def build_relationship_choices(answer_store, group_instance):
     :param group_instance: The instance of the group being iterated over
     :return:
     """
-    household_answers = answer_store.filter(block_id='household-composition')
+    household_first_name_answers = answer_store.filter(block_id='household-composition', answer_id='first-name', limit=settings.EQ_MAX_NUM_REPEATS)
```

Comment:
Assuming that change thats in another PR doesn't stop it storing blank entries, the trim thing?